content/articles/flow-and-typescript-1.md

```diff
@@ -0,0 +1,190 @@
+Category: Dev
+Authors: Vincent, Tom
+
+
+*Reading time: ~10 minutes.*
+
+As front-end applications grow in complexity, more and more developers are looking for ways to make development and maintenance more scalable.
+For some people it means foregoing JavaScript itself and using languages such as [Clojurescript](https://github.com/clojure/clojurescript), [Elm](http://elm-lang.org/) or [Purescript](http://www.purescript.org/) while others want to keep using JavaScript, especially now that ES6 makes writing it tolerable.
+
+A simple way to make a codebase easier to maintain and to grow is to have types and that is exactly what [Flow](https://babeljs.io/) and [TypeScript](http://www.typescriptlang.org/) offer, albeit in different ways.
+<!-- PELICAN_END_SUMMARY -->
+
+TypeScript is a ES6 transpiler like [Babel](https://babeljs.io/) with types while Flow only does the annotation and static type checking. If you want to use ES6 with Flow, you will need to use Babel as well.
+
+For this series of articles, I am going to convert my [react-example](https://github.com/Keats/react-example) which is a very basic react/redux/react-router Kanban board app to both Flow and TypeScript. Both TypeScript and Flow versions are in the same [repo](https://github.com/Keats/flow-typescript) but only the Flow one is working at the time of writing.
+
+In this part 1, we will look at Flow.
+
+## Installation and usage
+
+It's important to say that Flow doesn't currently work on Windows. You can have a look at [issue #6](https://github.com/facebook/flow/issues/6) on their tracker if you are using Windows. Running it in a [docker image](https://hub.docker.com/r/motiz88/flow/) might work for you though.
+
+For the sake of simplicity I will install the Flow binary with [flow-bin](https://www.npmjs.com/package/flow-bin).instead of using e.g. a gulp plugin (mainly since I couldn't find one).
+
+One nice feature in Flow is that the type checking is opt-in: only the files with the `/* @flow */` comment at the top of a file will be checked. This makes transitioning to Flow extremely easy: you can go one file at a time. Flow also has a `suggest` feature that finds a number of type annotations automatically. `suggest` prints a diff of the type annotations:
+
+```diff
+$ flow suggest src/app/store.js
+--- old
++++ new
+@@ -16,7 +16,7 @@
+   router: routerStateReducer,
+   ...reducers,
+ });
+-const middlewares = [thunk];
++const middlewares: Array<{thunk: () => any,}> = [thunk];
+```
+
+## An example
+
+Let's start with the [`routes.js`](https://github.com/Keats/flow-typescript/blob/master/flow/src/app/routes.js) files and we get the following error:
+
+```
+  5: import { Route, IndexRoute } from "react-router";
+                                       ^^^^^^^^^^^^^^ react-router. Required module not found
+
+```
+As you can see, Flow doesn't complain about the React import as it comes bundled with an interface file for it.
```

Comment:
I think it's not obvious that `routes.js` starts with `import React from "react";`, it took me a minute to re-read this